lib/medium-admin.php

```diff
@@ -796,16 +796,28 @@ public static function cross_post($post, $medium_post, $medium_user) {
         "Content-Type" => "application/json"
       ));
     } catch (Exception $e) {
-      // Retry once if we got a timeout
+      // Retry on the server failure response codes
+      $retry_response_codes = array(
+        500,
+        502,
+        503,
+        504
+      );
+
+      // Retry once for timeout or server error
       if ($e->getCode() == -2) {
         error_log("RETRYING POST $post->ID '$post->post_title' due to timeout, delaying...");
         sleep(5);
-        $created_medium_post = self::_medium_request("POST", $path, $medium_user->token, $data, array(
-          "Content-Type" => "application/json"
-        ));
+      } else if (in_array($e->getCode(), $retry_response_codes)) {
+        error_log("RETRYING POST $post->ID '$post->post_title' due to response code $code, delaying...");
+        sleep(5);
```

Comment:
Yeah, I thought about this for awhile. In general, this should never happen, but at scale, we have seen cases where this happens. My thinking about this is that retry once for now. It probably was a fluke if it happened, but, if it does happen a second time, then we probably want to stop retrying, because this indicates that there could be a much larger problem at hand, such as capacity constraints on a system, which means that we should stop retrying to exacerbate the problem.

WDYT?